extension/src/support.c

```diff
@@ -0,0 +1,64 @@
+/* This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE for a copy of the license
+ */
+
+#include <postgres.h>
+#include <fmgr.h>
+
+#include <nodes/pathnodes.h>
+#include <nodes/supportnodes.h>
```

Comment:
I think this should be documented significantly more, in terms of what it's doing, what its limitations are etc. It's very difficult to review this way and also will mean that it's much more difficult to maintain.